worktree.c

```diff
@@ -51,8 +51,8 @@ static struct worktree *get_main_worktree(void)
 	struct strbuf worktree_path = STRBUF_INIT;
 
 	strbuf_add_absolute_path(&worktree_path, get_git_common_dir());
-	if (!strbuf_strip_suffix(&worktree_path, "/.git"))
-		strbuf_strip_suffix(&worktree_path, "/.");
+	strbuf_strip_suffix(&worktree_path, "/.");
+ 	strbuf_strip_suffix(&worktree_path, "/.git");
```

Comment:
Oh, by the way, this is a bug fix that should be extracted into its own, "preparatory" commit. ("Preparatory" means in this case that it should be sorted to the beginning of your patch series, see https://git-scm.com/docs/git-rebase#_splitting_commits how to do that)